### PR TITLE
Call bf16 -> f16 conversion in NPUW

### PR DESCRIPTION
**Pull request by slyalin**
Calls ConvertPrecision transformation to convert bf16 to f16 for entire model as a model processing step inside NPUW plugin. Relies on the recently added fast bf16->f16 conversion in OV core.

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp [hunk 1]
```diff
@@ -127,6 +128,8 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
         }
     }
 
+    ov::pass::ConvertPrecision(ov::element::bf16, ov::element::f16).run_on_model(model);
```
Comment (smirnov-alexey) [inline src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:131]: Could you please put a `FIXME` here to find the best place for this transformation? I'm not 100% sure it should be here and run unconditionally

**Review by smirnov-alexey — APPROVED** [review]
LGTM

**Comment by smirnov-alexey** [comment]
Quick note: there is no need to `clone()` the model since it's already cloned for NPUW here https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_npu/src/plugin/src/plugin.cpp#L598

**Comment by smirnov-alexey** [comment]
I've tested the changes with `bf16` model and the subgraphs have their weights converted to `fp16`. Also our previously tested fp16 model work w/o changes

**Review by praasz — APPROVED** [review]

**Review by itikhono — APPROVED** [review]

**Review by TolyaTalamanov — APPROVED** [review]

**Pull request merged**